### DISCUSSION TITLE
Ability to store Infisical's environment variables directly in the database

### DISCUSSION THREAD

**Discussion by binaryben** [Infisical/infisical#5429] [Feature Request]
### Feature description

Provide a startup wizard or another option for configuring and securely storing credentials required by Infisical without requiring environment variables.

### Why would it be useful?

From a marketing point of view, it's ironic that Infisical is software that advocates against storing credentials in text files and then practically[^practical] requires that as part of its' own startup process (assuming the use of Docker for sake of a simple example). Ideally, I'd expect the startup process to require me to enter the encryption key manually and everything else is stored directly in the database with the secrets for other apps.

Specifically, I am hoping to use Infisical to store credentials such as SMTP provider auth which can be used by a bad actor to phish for user credentials.

[^practical]: I could, for example, use the free tier of Infisical Cloud offering to inject the secrets. That would defeat some of the benefits of self-hosting though from my point of view. I could also write a simple script to prompt for the credentials before starting up, but it seems impractical to do at every restart when I am launching an entire system to securely store credentials.

**Comment by dangtony98** [comment]
This is an interesting idea but definitely a big change for the codebase as we'd have to move many variables inside @binaryben 

I'd have to discuss this with the team and go from there — considering many ongoing [initiatives](https://www.notion.so/infisical/be2d2585a6694e40889b03aef96ea36b?v=5b19a8127d1a4060b54769567a8785fa) and our limited bandwidth/capacity, I wouldn't expect this for some time though. That said, I'd be happy to continue the discussion here.

**Comment by repelliuss** [comment]
I was also surprised by this. I am curious if there is another way without requiring environment variables to setup infisical as of now

**Comment by akhilmhdh** [comment]
We do now support partially configuration like integration environment variable to be in server admin console. Infrastructure environment variable can be configured like this way and environment variable is more simpler approach for that. 


Moving this to discussions as it's a feature request
